pkg/scheduler/preemption/preemption.go

```diff
@@ -248,7 +248,7 @@ func findCandidates(wl *kueue.Workload, cq *cache.ClusterQueue, resPerFlv resour
 	}
 
 	if cq.Cohort != nil && cq.Preemption.ReclaimWithinCohort != kueue.PreemptionPolicyNever {
-		cqs := cq.Cohort.Members
+		cqs := cq.Cohort.Members.Clone()
```

Comment:
nit: I think without degrading readability we could do `if cohortCq == cq { continue }` below, instead of the Clone and Delete.

Reply:
I'm also wondering if we have the exclusion of the `cq` here covered with unit tests. If not, it is worth a follow up as part of improving unit test coverage.

Reply:
Changed to `continue`.

Yes, we have it covered (every test for `ReclaimWithinCohort` would need to exclude the CQ itself)